### PR TITLE
Pcvl 788 use post selected cnot as much as possible in converters

### PR DESCRIPTION
**Pull request by raksharuia**

### CODE REVIEW

**Review by MarionQuandela — CHANGES_REQUESTED** [review]

File: perceval/converters/myqlm_converter.py [hunk 1]
```diff
@@ -60,7 +70,10 @@ def convert(self, qlmc, use_postselection: bool = True) -> Processor:
         # this nested import fixes automatic class reference generation
 
         get_logger().info(f"Convert myQLM circuit ({qlmc.nbqbits} qubits) to processor", channel.general)
-        n_cnot = qlmc.count("CNOT")  # count the number of CNOT gates in circuit - needed to find the num. heralds
```
Comment (ericbrts) [inline perceval/converters/myqlm_converter.py:63]: Please update docstring, as done in the `QiskitConverter`

File: perceval/converters/abstract_converter.py [hunk 2]
```diff
@@ -52,8 +52,7 @@ class AGateConverter(ABC):
     def __init__(self, backend_name: str = "SLOS", source: Source = Source()):
         self._converted_processor = None
         self._input_list = None  # input state in list
-        self._cnot_idx = 0  # counter for CNOTS in circuit
-        self._source = source
+        self._noise_model = NoiseModel()
```
Comment (ericbrts) [inline perceval/converters/abstract_converter.py:55]: You're replacing `Source` with `NoiseModel` which is cool, but please go to the end of it:
* deprecated the source usage
* have a `noise` parameter in the `__init__` of all converters.

File: perceval/converters/abstract_converter.py [hunk 3]
```diff
@@ -49,10 +49,14 @@ class AGateConverter(ABC):
     Converter class for gate based Circuits to perceval processor
     """
 
-    def __init__(self, backend_name: str = "SLOS", source: Source = Source()):
+    def __init__(self, backend_name: str = "SLOS", source: Source = None, noise_model: NoiseModel = NoiseModel()):
         self._converted_processor = None
         self._input_list = None  # input state in list
-        self._noise_model = NoiseModel()
+        if source is not None:
```
Comment (ericbrts) [inline perceval/converters/abstract_converter.py:55]: Ok, but here `self._source` is never used.

I'd suggest:
```python
self._noise_model = NoiseModel()
if source is not None:
    get_logger().warn('DeprecationWarning: Call with deprecated argument "source", '
                              'please use "noise_model=NoiseModel()" instead')
    self._noise_model = NoiseModel(transmittance=1-source._losses,
                                   brightness=source._emission_probability,
                                   g2=source._multiphoton_component,
                                   indistinguishability=source._indistinguishability,
                                   g2_distinguishable=(source._multiphoton_model=='distinguishable')
                                   )
if noise_model is not None:
    self._noise_model = noise_model
```
And set the default value of `noise_model` to `None` too

File: perceval/converters/cqasm_converter.py [hunk 4]
```diff
@@ -86,8 +87,8 @@ class CQASMConverter(AGateConverter):
     :param backend_name: backend name used in the converted processor (default SLOS)
     :param source: the source used as input for the converted processor (default perfect source).
     """
-    def __init__(self, backend_name: str = "SLOS", source: Source = Source()):
-        super().__init__(backend_name, source)
+    def __init__(self, backend_name: str = "SLOS", source: Source = Source(), noise_model: NoiseModel = NoiseModel()):
```
Comment (ericbrts) [inline perceval/converters/cqasm_converter.py:90]: In all initializers, the default for `source` and `noise_model` should be `None`

File: perceval/converters/abstract_converter.py (hunk 3 above)
Comment (raksharuia) [reply perceval/converters/abstract_converter.py:55]: You are absolutely correct, i completely forgot that my update would render the source unused. Fixed.

**Pull request merged**